### PR TITLE
Fix: Branch `main` could not be tested easily

### PR DESCRIPTION
**Pull request by hoh**
This deploys the main branch automatically on the staging servers for system testing.

### CODE REVIEW

**Comment by olethanh** [comment]
both server are on debian 12? Wouldn't it make sense to have one on ubuntu?

**Review by olethanh — APPROVED** [review]
LGTM.
I trust you have tested it somehow

File: .github/workflows/deploy-main-on-staging.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# This workflow automatically deploys main on staging
+name: "Deploy `main` automatically on staging"
+
+on:
+  push:
+    branches:
+      - main
```
Comment (nesitor) [inline .github/workflows/deploy-main-on-staging.yml:7]: Maybe we can do it using a tag instead automatically? Just to have better control on it.

**Comment by hoh** [comment]
> both server are on debian 12? Wouldn't it make sense to have one on ubuntu?

Yes, that would make sense. However, would it then be difficult to know if a bug comes from the OS or from the hosting provider :thinking: ?

File: .github/workflows/deploy-main-on-staging.yml (hunk 1 above)
Comment (hoh) [reply .github/workflows/deploy-main-on-staging.yml:7]: I think automatically would make things easier: we'll always know that the latest version runs there.

**Pull request merged**